contracts/ism-multisig/src/query.rs

```diff
@@ -32,34 +29,31 @@ pub fn verify_message(
         signatures.push(metadata.signature_at(i))
     }
 
-    let unique_vali_pubkey: HashSet<_> = validators
+    let verifiable_cases = validators
         .0
         .into_iter()
-        .map(|v| uncompress_pubkey(v.signer_pubkey).unwrap())
-        .collect();
+        .map(|v| {
+            signatures
+                .clone()
+                .into_iter()
+                .map(|s| (v.signer_pubkey.clone(), s))
+                .collect::<Vec<(Binary, Binary)>>()
+        })
+        .fold(Vec::<(Binary, Binary)>::new(), |acc, item| {
+            acc.into_iter().chain(item.into_iter()).collect()
+        });
 
-    let unique_meta_pubkey: HashSet<_> = signatures
+    let success: u8 = verifiable_cases
         .into_iter()
-        .flat_map(|sig| {
-            [
-                deps.api
-                    .secp256k1_recover_pubkey(&message.id(), &sig[0..64], 0)
-                    .unwrap(),
-                deps.api
-                    .secp256k1_recover_pubkey(&message.id(), &sig[0..64], 1)
-                    .unwrap(),
-            ]
+        .map(|v| {
+            deps.api
+                .secp256k1_verify(&message.id(), &v.1[0..64], &v.0)
```

Comment:
how do we guarantee the v's length is shorter than 64?

Reply:
because the origin of v signature - `metadata.signature_at` try to slice bytes as length 65. @byeongsu-hong

Reply:
:+1: